tasks/main.yml

```diff
@@ -34,12 +34,32 @@
     - name: Gather OpenStack infrastructure information
       command: openstack stack output show {{ cluster_name }} cluster_group -f json
       register: stack_output
+      changed_when: false
 
     - name: Extract node groups
       set_fact:
         cluster_group: "{{ stack_output.stdout | from_json }}"
   when: cluster_state == 'query'
 
+- name: Extract server ids from cluster group output
+  set_fact:
+    cluster_servers: "{{ cluster_group.output_value | json_query('[*].nodes[*].id') | flatten }}"
+    cluster_interfaces: "{{ cluster_net | map(attribute='net') | list }}"
+
+- name: Attach interfaces to servers
+  os_server_interface:
+    auth_type: "{{ cluster_auth_type or omit }}"
+    auth: "{{ cluster_auth or omit }}"
+    cloud: "{{ cluster_cloud or omit }}"
+    state: "{{ cluster_state }}"
+    server_id: "{{ item }}"
+    interfaces: "{{ cluster_interfaces }}"
+  with_items: "{{ cluster_servers }}"
+  register: openstack_server_interfaces
```

Comment:
Does this need when: cluster_state in ['present', 'query'] ?

Reply:
Yep, in query state, it doesn't add new interfaces, only gets a list of interfaces attached to a server.